lib/rules/lines-around-comment.js

```diff
@@ -264,7 +264,8 @@ module.exports = {
                 isCommentAtParentStart(token, "ClassBody") ||
                 isCommentAtParentStart(token, "BlockStatement") ||
                 isCommentAtParentStart(token, "StaticBlock") ||
-                isCommentAtParentStart(token, "SwitchCase")
+                isCommentAtParentStart(token, "SwitchCase") ||
+                isCommentAtParentStart(token, "SwitchStatement")
```

Comment:
Since `SwitchStatement` is not special-cased in functions that calculate this, they'll treat `switch` as the block start but it should be `{`.

```js
/* eslint lines-around-comment: ["error", {
    allowBlockStart: true,
    beforeLineComment: true,
    afterLineComment: true
}] */

switch (
    // this comment is a false negative

    a
) {
    case foo:
        break;
}

switch (a)
{
    // this comment is a false positive

    case foo:
        break;
}

```

Reply:
Fixed and added test cases.